addons/web/static/src/js/views/list/list_editable_renderer.js

```diff
@@ -404,7 +408,18 @@ ListRenderer.include({
      * @returns {Promise}
      */
     setRowMode: function (recordID, mode) {
-        var self = this;
+        // Use a mutex because we don't want to rerender a row before the previous render is finished
+        let mutex = this.rowModeMutexesByRecordId.get(recordID);
+        if (!mutex) {
+            mutex = new concurrency.Mutex();
+            this.rowModeMutexesByRecordId.set(recordID, mutex);
+        }
+
+        return mutex.exec(this._setRowMode.bind(this, recordID, mode));
+    },
+
+    _setRowMode: async function (recordID, mode) {
```

Comment:
wouldn't it be easier (and more robust) to have a single mutex for all row mode changes, whatever the record id?

Reply:
Of course it will slow things down